### ISSUE TITLE
Fail to download Private RefTimeSeries via rest api 

### ISSUE THREAD

**Issue by zhiyuli** [hydroshare/hydroshare#1363]
A "401 Unauthorized access" error occurs when retrieving a private RefTS resource via rest client after logging in through either basic authorization or OAuth.

@pkdash Looks like this change I made does not work for RefTS.  Any suggestion? thanks
https://github.com/hydroshare/hydroshare/blob/develop/hs_core/views/resource_rest_api.py#L264

**Comment by pkdash** [comment]
@zhiyuli Try not including ```site_url``` when you generating ```bag_url``` for the ref timeseries (the above line in the link you have provided)

**Comment by zhiyuli** [comment]
@pkdash I tried the change you suggested and it is still not working.....
 https://github.com/hydroshare/hydroshare/commit/50014b11e05441c23fca1c8fc577b3ee0c3611f4

I deployed the change on park and shared a private refts res with you
https://park.hydroshare.org/resource/ce943b09e93e4dea84d3e3d521c538fc/

**Comment by zhiyuli** [comment]
@pkdash the code I used for testing is here:
from hs_restclient import HydroShare, HydroShareAuthBasic
hs_hostname = "park.hydroshare.org"
auth = HydroShareAuthBasic(username='**USER_NAME**', password='**USER_PASSWORD**')
hs = HydroShare(auth=auth, hostname=hs_hostname)
hs.getResource("ce943b09e93e4dea84d3e3d521c538fc", destination="/tmp", unzip=True)

**Comment by pkdash** [comment]
Can you try using this decorator ```@api_view(['GET'])``` for the ref_ts fie download view function? Also, add the ```site_url``` if you removed as I suggested previously. Here is a reference  if you need how this decorator has been used.
https://github.com/hydroshare/django_irods/blob/911eee0ee7a57001273255286c19e4f308322373/views.py#L23

**Comment by zhiyuli** [comment]
@pkdash adding @api_view(['GET']) works! I deployed the change on park. 

I found one thing: rest api cannot be properly tested on my local hs vm. 
I wrote a simple python script using hs_restclient to test against my local hs vm first. When trying downloading private generic and refts resource bags, I got 401 error on both.
But when running the same script against park, it works on both.
I am not sure why the different deployments (local v.s.  remote testing server) can cause this difference.
